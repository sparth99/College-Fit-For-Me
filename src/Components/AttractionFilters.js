import React from 'react';
import Button from 'react-bootstrap/Button';
import axios from 'axios';
import { Dropdown, ButtonToolbar } from 'react-bootstrap';

class AttractionFilters extends React.Component {
  state = {
    attractions: null,
    state_filter: 'Select State',
    price_filter: 'Select Price',
    sort: 'Sort By',
    order: 'Order',
    apiLink: '',
  };

  constructor(props) {
    super(props);
  }

  state_selected = (state_name) => {
    return state_name !== 'Select State' && state_name !== 'All';
  };

  price_selected = (price) => {
    return price !== 'Select Price' && price !== 'All';
  };

  sort_selected = (sort) => {
    return sort !== 'Sort By' && sort !== 'All';
  };

  render() {
    return (
      <ButtonToolbar className="CitiesFilter">
        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.state_filter}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {[
              'All',
              'Alabama',
              'Alaska',
              'Arizona',
              'Arkansas',
              'California',
              'Colorado',
              'Connecticut',
              'Delaware',
              'Florida',
              'Georgia',
              'Hawaii',
              'Idaho',
              'Illinois',
              'Indiana',
              'Iowa',
              'Kansas',
              'Kentucky',
              'Louisiana',
              'Maine',
              'Maryland',
              'Massachusetts',
              'Michigan',
              'Minnesota',
              'Mississippi',
              'Missouri',
              'Montana',
              'Nebraska',
              'Nevada',
              'New Hampshire',
              'New Jersey',
              'New Mexico',
              'New York',
              'North Carolina',
              'North Dakota',
              'Ohio',
              'Oklahoma',
              'Oregon',
              'Pennsylvania',
              'Rhode Island',
              'South Carolina',
              'South Dakota',
              'Tennessee',
              'Texas',
              'Utah',
              'Vermont',
              'Virginia',
              'Washington',
              'West Virginia',
              'Wisconsin',
              'Wyoming',
            ].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ state_filter: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.price_filter}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {['All', '$', '$$', '$$$', '$$$$'].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ price_filter: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.sort}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {['All', 'Rating', 'Review Count'].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ sort: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.order}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {['Ascending', 'Descending'].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ order: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Button
          className="m-1"
          id="go-btn"
          variant="outline-primary"
          onClick={() => {
            if (
              this.state_selected(this.state.state_filter) &&
              this.price_selected(this.state.price_filter) &&
              this.sort_selected(this.state.sort)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                    this.state.state_filter +
                    '&price=' +
                    this.state.price_filter +
                    '&order=' +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*dsc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                        this.state.state_filter +
                        '&price=' +
                        this.state.price_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                        this.state.state_filter +
                        '&price=' +
                        this.state.price_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else if (
              this.state_selected(this.state.state_filter) &&
              this.price_selected(this.state.price_filter)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                    this.state.state_filter +
                    '&price=' +
                    this.state.price_filter +
                    '&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(
                    `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                      this.state.state_filter +
                      '&price=' +
                      this.state.price_filter +
                      '&page=1'
                  );
                });
            } else if (
              this.state_selected(this.state.state_filter) &&
              this.sort_selected(this.state.sort)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                    this.state.state_filter +
                    '&order=' +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*dsc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                        this.state.state_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                        this.state.state_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else if (this.state_selected(this.state.state_filter)) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                    this.state.state_filter +
                    '&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(
                    `https://api.collegefitfor.me/restaurants/filter/?state_name=` +
                      this.state.state_filter +
                      '&page=1'
                  );
                });
            } else if (
              this.price_selected(this.state.price_filter) &&
              this.sort_selected(this.state.sort)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?price=` +
                    this.state.price_filter +
                    '&order=' +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*dsc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?price=` +
                        this.state.price_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?price=` +
                        this.state.price_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else if (this.price_selected(this.state.price_filter)) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?price=` +
                    this.state.price_filter +
                    '&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(
                    `https://api.collegefitfor.me/restaurants/filter/?price=` +
                      this.state.price_filter +
                      '&page=1'
                  );
                });
            } else if (this.sort_selected(this.state.sort)) {
              axios
                .get(
                  `https://api.collegefitfor.me/restaurants/filter/?order=` +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*dsc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?order=` +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/restaurants/filter/?order=` +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else {
              axios
                .get(`https://api.collegefitfor.me/restaurants`)
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(`https://api.collegefitfor.me/restaurants`);
                });
            }
          }}
        >
          Go
        </Button>
      </ButtonToolbar>
    );
  }
}

export default AttractionFilters;
