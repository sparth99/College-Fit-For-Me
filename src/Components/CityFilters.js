import React from 'react';
import Button from 'react-bootstrap/Button';
import axios from 'axios';
import { Dropdown, ButtonToolbar } from 'react-bootstrap';

class CityFilters extends React.Component {
  state = {
    cities: null,
    state_filter: 'Select State',
    time_filter: 'Select Time Zone',
    sort: 'Sort By',
    order: 'Order',
    apiLink: '',
  };

  constructor(props) {
    super(props);
  }

  state_selected = (state_name) => {
    return state_name !== 'Select State' && state_name !== 'All';
  };

  time_selcted = (time_zone) => {
    return time_zone !== 'Select Time Zone' && time_zone !== 'All';
  };

  sort_selected = (sort) => {
    return sort !== 'Sort By' && sort !== 'All';
  };

  time_zone_value = (time_zone) => {
    if (time_zone === 'Central Time') {
      return 'America/Chicago';
    } else if (time_zone === 'Mountain Time') {
      // Missing Boise
      return 'America/Denver';
    } else if (time_zone === 'Mountain Time w/o DST') {
      return 'America/Phoenix';
    } else if (time_zone === 'Pacific Time') {
      return 'America/Los_Angeles';
    } else if (time_zone === 'Alaska Time') {
      return 'America/Anchorage';
    } else if (time_zone === 'Hawaii-Aleutian Time') {
      return 'Pacific/Honolulu';
    } else {
      // Missing Detroit, Indianapolis and Louisville
      return 'America/New_York';
    }
  };

  render() {
    return (
      <ButtonToolbar className="CitiesFilter">
        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.state_filter}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {[
              'All',
              'Alabama',
              'Alaska',
              'Arizona',
              'Arkansas',
              'California',
              'Colorado',
              'Connecticut',
              'Delaware',
              'Florida',
              'Georgia',
              'Hawaii',
              'Idaho',
              'Illinois',
              'Indiana',
              'Iowa',
              'Kansas',
              'Kentucky',
              'Louisiana',
              'Maine',
              'Maryland',
              'Massachusetts',
              'Michigan',
              'Minnesota',
              'Mississippi',
              'Missouri',
              'Montana',
              'Nebraska',
              'Nevada',
              'New Hampshire',
              'New Jersey',
              'New Mexico',
              'New York',
              'North Carolina',
              'North Dakota',
              'Ohio',
              'Oklahoma',
              'Oregon',
              'Pennsylvania',
              'Rhode Island',
              'South Carolina',
              'South Dakota',
              'Tennessee',
              'Texas',
              'Utah',
              'Vermont',
              'Virginia',
              'Washington',
              'West Virginia',
              'Wisconsin',
              'Wyoming',
            ].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ state_filter: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.time_filter}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {[
              'All',
              'Eastern Time',
              'Central Time',
              'Mountain Time',
              'Mountain Time w/o DST',
              'Pacific Time',
              'Alaska Time',
              'Hawaii-Aleutian Time',
            ].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ time_filter: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.sort}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {[
              'All',
              'Population',
              'Crime Index',
              'Pollution Index',
              'Rent Index',
              'Restaurant Price Index',
              'Traffic Index',
              'Safety Index',
              'Health Care Index',
            ].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ sort: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Dropdown className="m-1" id="dropdown-btn">
          <Dropdown.Toggle variant="success" id="dropdown-basic" variant="dark">
            {this.state.order}
          </Dropdown.Toggle>
          <Dropdown.Menu className="ourdropdown" id="dropdown-options">
            {['Ascending', 'Descending'].map((value) => {
              return (
                <Dropdown.Item
                  key={value}
                  onClick={() => {
                    this.setState({ order: value });
                  }}
                >
                  {value}
                </Dropdown.Item>
              );
            })}
          </Dropdown.Menu>
        </Dropdown>

        <Button
          className="m-1"
          id="go-btn"
          variant="outline-primary"
          onClick={() => {
            if (
              this.state_selected(this.state.state_filter) &&
              this.time_selcted(this.state.time_filter) &&
              this.sort_selected(this.state.sort)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?state_name=` +
                    this.state.state_filter +
                    '&time_zone=' +
                    this.time_zone_value(this.state.time_filter) +
                    '&order=' +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*asc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?state_name=` +
                        this.state.state_filter +
                        '&time_zone=' +
                        this.time_zone_value(this.state.time_filter) +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?state_name=` +
                        this.state.state_filter +
                        '&time_zone=' +
                        this.time_zone_value(this.state.time_filter) +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else if (
              this.state_selected(this.state.state_filter) &&
              this.time_selcted(this.state.time_filter)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?state_name=` +
                    this.state.state_filter +
                    '&time_zone=' +
                    this.time_zone_value(this.state.time_filter) +
                    '&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(
                    `https://api.collegefitfor.me/cities/filter/?state_name=` +
                      this.state.state_filter +
                      '&time_zone=' +
                      this.time_zone_value(this.state.time_filter) +
                      '&page=1'
                  );
                });
            } else if (
              this.state_selected(this.state.state_filter) &&
              this.sort_selected(this.state.sort)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?state_name=` +
                    this.state.state_filter +
                    '&order=' +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*asc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?state_name=` +
                        this.state.state_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?state_name=` +
                        this.state.state_filter +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else if (this.state_selected(this.state.state_filter)) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?state_name=` +
                    this.state.state_filter +
                    '&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(
                    `https://api.collegefitfor.me/cities/filter/?state_name=` +
                      this.state.state_filter +
                      '&page=1'
                  );
                });
            } else if (
              this.time_selcted(this.state.time_filter) &&
              this.sort_selected(this.state.sort)
            ) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?time_zone=` +
                    this.time_zone_value(this.state.time_filter) +
                    '&order=' +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*asc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?time_zone=` +
                        this.time_zone_value(this.state.time_filter) +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?time_zone=` +
                        this.time_zone_value(this.state.time_filter) +
                        '&order=' +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else if (this.time_selcted(this.state.time_filter)) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?time_zone=` +
                    this.time_zone_value(this.state.time_filter) +
                    '&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  this.props.setter(
                    `https://api.collegefitfor.me/cities/filter/?time_zone=` +
                      this.time_zone_value(this.state.time_filter) +
                      '&page=1'
                  );
                });
            } else if (this.sort_selected(this.state.sort)) {
              axios
                .get(
                  `https://api.collegefitfor.me/cities/filter/?order=` +
                    this.state.sort.toLowerCase().replace(/ /g, '_') +
                    '*asc&page=1'
                )
                .then((res) => {
                  const attractions = res.data;
                  this.setState({ attractions });
                  if (this.state.order === 'Descending') {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?order=` +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*dsc&page=1'
                    );
                  } else {
                    this.props.setter(
                      `https://api.collegefitfor.me/cities/filter/?order=` +
                        this.state.sort.toLowerCase().replace(/ /g, '_') +
                        '*asc&page=1'
                    );
                  }
                });
            } else {
              axios.get(`https://api.collegefitfor.me/cities`).then((res) => {
                const attractions = res.data;
                this.setState({ attractions });
                this.props.setter(`https://api.collegefitfor.me/cities`);
              });
            }
          }}
        >
          Go
        </Button>
      </ButtonToolbar>
    );
  }
}

export default CityFilters;
