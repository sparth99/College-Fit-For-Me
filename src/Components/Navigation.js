import React from 'react';
import SearchInput from './SearchInput.js';
import { Nav, Navbar } from 'react-bootstrap';

class Navigation extends React.Component {
  render() {
    return (
      <>
        <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
          <Navbar.Brand href="/home">College Fit For Me</Navbar.Brand>
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse id="responsive-navbar-nav">
            <Nav className="mr-auto">
              <Nav.Link href="/cities">Cities</Nav.Link>
              <Nav.Link href="/universities">Universities</Nav.Link>
              <Nav.Link href="/attractions">Attractions</Nav.Link>
              <Nav.Link href="/visualizations">Visualizations</Nav.Link>
              <Nav.Link href="/about">About</Nav.Link>
            </Nav>
            <SearchInput />
          </Navbar.Collapse>
        </Navbar>
      </>
    );
  }
}

export default Navigation;
