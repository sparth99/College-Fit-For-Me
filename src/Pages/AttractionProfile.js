import React, { Component } from 'react';
import '../Styles/Profile.css';
import AttractionData from './AttractionData';
import axios from 'axios';

class AttractionProfile extends Component {
  state = {
    attraction: null,
  };

  componentDidMount() {
    const tokenized_url = String(window.location.href).split('/');
    const attraction_name = tokenized_url[tokenized_url.length - 1]
      .replace(/_/g, '%20')
      .replace(/&/g, '%26');
    axios
      .get('https://api.collegefitfor.me/restaurants?name=' + attraction_name)
      .then((res) => {
        var attraction = res.data;
        this.setState({ attraction });
      });
  }

  render() {
    if (this.state.attraction === null) {
      return <b> Loading ... </b>;
    } else {
      return (
        <div>
          <AttractionData attr={this.state.attraction[0]} />
        </div>
      );
    }
  }
}

export default AttractionProfile;
