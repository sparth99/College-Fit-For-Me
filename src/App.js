import React, { Component } from 'react';
import './App.css';
import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
import { Route, BrowserRouter } from 'react-router-dom';
import Home from './Pages/Home.js';
import About from './Pages/About.js';
import Cities from './Pages/Cities.js';
import CityProfile from './Pages/CityProfile.js';
import Universities from './Pages/Universities.js';
import UniversityProfile from './Pages/UniversityProfile.js';
import Visualizations from './Pages/Visualizations.js';
import Attractions from './Pages/Attractions.js';
import AttractionProfile from './Pages/AttractionProfile.js';
import SearchResults from './Pages/SearchResults.js';

class App extends Component {
  render() {
    return (
      <div className="App">
        <BrowserRouter>
          <Route exact path={'/'} component={Home} />
          <Route exact path={'/universities'} component={Universities} />
          <Route
            path={'/universities/:universityname'}
            component={UniversityProfile}
          />
          <Route path={'/about'} component={About} />
          <Route path={'/home'} component={Home} />
          <Route path={'/visualizations'} component={Visualizations} />
          <Route exact path={'/attractions'} component={Attractions} />
          <Route
            path={'/attractions/:attractionname'}
            component={AttractionProfile}
          />
          <Route exact path={'/cities'} component={Cities} />
          <Route path={`/cities/:cityname`} component={CityProfile} />
          <Route path={`/search/:modelType`} component={SearchResults} />
        </BrowserRouter>
      </div>
    );
  }
}
export default App;
