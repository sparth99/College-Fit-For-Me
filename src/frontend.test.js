import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, configure} from 'enzyme';	
import Adapter from 'enzyme-adapter-react-16';
import App from './App';
import Navigation from './Components/Navigation';
import Home from './Pages/Home';
import Cities from './Pages/Cities.js'
import Universities from "./Pages/Universities";
import Attractions from "./Pages/Attractions";
import CityProfile from "./Pages/CityProfile";
import UniversityProfile from "./Pages/UniversityProfile";
import AttractionProfile from "./Pages/AttractionProfile";
import AttractionFilters from "./Components/AttractionFilters";
import CityFilters from "./Components/CityFilters";
import UniversityFilters from "./Components/UniversityFilters";
import GitLab from "./Components/Gitlab";
import About from './Pages/About';

configure({ adapter: new Adapter() });

it('should render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Navigation Component', () => {
	it('should render a Navigation Bar', () => {
	  const nav = shallow(<Navigation />);

      expect(nav).toMatchSnapshot();
	  
	});
});

describe("Home Component", () => {
  const component = shallow(<Home/>);
  it('should render title', () => {
    expect(component.find('h1').at(0).text()).toEqual("Hello Students!")
  })
  it('should render description', () => {
    expect(component.find('p').at(0).text()).toEqual("Find the University that\'s the best fit for your needs and interests!Learn about life at different Universities, their surrounding areas, and explore attractions near them.")
  })
})

it('should render Cities', async () => {
  const component = shallow(<Cities/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();	
  
});

it('should render Universities', async () => {
  const component = shallow(<Universities/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();

});

it('should render Attractions', async () => {
  const component = shallow(<Attractions/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();
  
});

it('should render CityProfile', async () => {
  const component = shallow(<CityProfile/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();	
  
});

it('should render UniversityProfile', async () => {
  const component = shallow(<UniversityProfile/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();

});

it('should render AttractionProfile', async () => {
  const component = shallow(<AttractionProfile/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();
  
});
  
it('should render About', async () => {
  const component = shallow(<About/>);
  expect(component).toMatchSnapshot();
});

describe("About Component", () => {
  const component = shallow(<About/>);
  it('should render title', () => {
    expect(component.find('h1').at(0).text()).toEqual("About Us")
  })
  it('should render mission statement', () => {
    expect(component.find('h3').at(0).text()).toEqual("We are a group of developers passionate about helping students!")
  })
  it('should render data sources', () => {
    expect(component.find('h2').at(0).text()).toEqual("Data Sources/API")
  })
  it('should render university data', () => {
    expect(component.find('h3').at(1).text()).toEqual("University Data")
  })
  it('should render attraction data', () => {
    expect(component.find('h3').at(3).text()).toEqual("Attraction Data")
  })
  it('should render city data', () => {
    expect(component.find('h3').at(5).text()).toEqual("City Data")
  })
  it('should render tools', () => {
    expect(component.find('h2').at(1).text()).toEqual("Tools")
  })
  it('should render links', () => {
    expect(component.find('h2').at(2).text()).toEqual("Links")
  })
})

it('should render GitLabTable', async () => {
  const component = shallow(<GitLab/>);
  const data = component.instance();
  await data.componentDidMount();
  expect(component).toMatchSnapshot();
	
});

it('should render AttractionFilters', async () => {
  const component = shallow(<AttractionFilters/>);
  expect(component).toMatchSnapshot();
  }
);

it('should render CityFilters', async () => {
  const component = shallow(<CityFilters/>);
  expect(component).toMatchSnapshot();
  }
);

it('should render UniversityFilters', async () => {
  const component = shallow(<UniversityFilters/>);
  expect(component).toMatchSnapshot();
  }
);

describe("Attractions Test", function() {
  let querystring = {
    pathname: "/attractions",
    search: "?page=1"
  };

  it("Attractions doesn't have any data at first render", function() {
    const root = shallow(<Attractions location={querystring} />);
    expect(root.state("attractions") == null);
  });
})

describe("Cities Test", function() {
  let querystring = {
    pathname: "/cities",
    search: "?page=1"
  };

  it("Cities doesn't have any data at first render", function() {
    const root = shallow(<Cities location={querystring} />);
    expect(root.state("cities") == null);
  });
})

describe("Universities Test", function() {
  let querystring = {
    pathname: "/universities",
    search: "?page=1"
  };

  it("Universities doesn't have any data at first render", function() {
    const root = shallow(<Universities location={querystring} />);
    expect(root.state("universities") == null);
  });
})



  
