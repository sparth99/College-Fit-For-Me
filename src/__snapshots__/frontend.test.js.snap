// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render a Navigation Bar 1`] = `ShallowWrapper {}`;

exports[`should render About 1`] = `ShallowWrapper {}`;

exports[`should render AttractionFilters 1`] = `ShallowWrapper {}`;

exports[`should render AttractionProfile 1`] = `ShallowWrapper {}`;

exports[`should render Attractions 1`] = `ShallowWrapper {}`;

exports[`should render Cities 1`] = `ShallowWrapper {}`;

exports[`should render CityFilters 1`] = `ShallowWrapper {}`;

exports[`should render CityProfile 1`] = `ShallowWrapper {}`;

exports[`should render GitLabTable 1`] = `ShallowWrapper {}`;

exports[`should render Universities 1`] = `ShallowWrapper {}`;

exports[`should render UniversityFilters 1`] = `ShallowWrapper {}`;

exports[`should render UniversityProfile 1`] = `ShallowWrapper {}`;
